pkg/scheduler/invoker/remoteinvoker.go

```diff
@@ -0,0 +1,186 @@
+package invoker
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/kyma-incubator/reconciler/pkg/model"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler"
+	"github.com/kyma-incubator/reconciler/pkg/scheduler/config"
+	"github.com/kyma-incubator/reconciler/pkg/scheduler/reconciliation"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+)
+
+const callbackURLTemplate = "%s://%s:%d/v1/operations/%s/callback/%s"
+
+type RemoteReconcilerInvoker struct {
+	reconRepo reconciliation.Repository
+	config    *config.Config
+	logger    *zap.SugaredLogger
+}
+
+func NewRemoteReoncilerInvoker(reconRepo reconciliation.Repository, cfg *config.Config, logger *zap.SugaredLogger) *RemoteReconcilerInvoker {
+	return &RemoteReconcilerInvoker{
+		reconRepo: reconRepo,
+		config:    cfg,
+		logger:    logger,
+	}
+}
+
+func (i *RemoteReconcilerInvoker) Invoke(_ context.Context, params *Params) error {
```

Comment:
Instead of just ignoring the context here, we can send the HTTP request with context. What do you think?

Reply:
aah, you mean that it's cancelling the http-call when context got closed?

Reply:
yes